app/webpacker/controllers/edit_variant_controller.js

```diff
@@ -160,13 +169,21 @@ export default class EditVariantController extends Controller {
     this.element.querySelector('[id="variant_unit_price"]').value = unit_price;
   }
 
+  #hideWeight() {
+    this.weight = this.element.querySelector('[id="variant_weight"]');
+    this.weight.parentElement.style.display = "none";
+  }
+
   #toggleWeight() {
-    let display = "block";
     if (this.variantUnit.value === "weight") {
-      display = "none";
+      return this.#hideWeight();
     }
 
+    // Show weight
     this.weight = this.element.querySelector('[id="variant_weight"]');
-    this.weight.parentElement.style.display = display;
+    this.weight.parentElement.style.display = "block";
```

Comment:
Shouldn't this be conditional? it seems to overwrite whatever hideWeight did.

Reply:
If we are hiding this bit of code doesn't run, see the return in the code above:
```javascript 
    if (this.variantUnit.value === "weight") {
      return this.#hideWeight();
    }

```